keel-ec2-api/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterSpec.kt

```diff
@@ -124,6 +126,35 @@ data class ClusterSpec(
   private val _artifactName: String? = null, // Custom backing field for artifactName, used by resolvers
   override val artifactVersion: String? = null
 ) : ComputeResourceSpec, Monikered, Locatable<SubnetAwareLocations>, OverrideableClusterDependencyContainer<ServerGroupSpec>, UnhappyControl {
+  @Factory
+  constructor(
+    moniker: Moniker,
+    imageProvider: ImageProvider? = null,
+    deployWith: ClusterDeployStrategy = RedBlack(),
+    @Optional locations: SubnetAwareLocations,
+    launchConfiguration: LaunchConfigurationSpec? = null,
+    capacity: Capacity? = null,
+    dependencies: ClusterDependencies? = null,
+    health: HealthSpec? = null,
+    scaling: Scaling? = null,
+    tags: Map<String, String>? = null,
+    overrides: Map<String, ServerGroupSpec> = emptyMap()
+  ) : this(
+    moniker,
+    imageProvider,
+    deployWith,
+    locations,
+    ServerGroupSpec(
+      launchConfiguration,
+      capacity,
+      dependencies,
+      health,
+      scaling,
+      tags
+    ),
+    overrides
+  )
```

Comment:
Secondary constructor for `ClusterSpec` without needing a custom deserializer or Jackson annotation